src/gretel_client/client.py

```diff
@@ -561,6 +566,20 @@ def install_packages(self, verbose: bool = False):
         pkg.install_packages(self.api_key, self.host, verbose)
 
 
+def _get_or_prompt(
+    input_key: str, prompt_message: str, env_fallback: str
+) -> Optional[str]:
+    """Helper function used to prompt for secrets based on env conditions."""
+    if input_key == PROMPT_ALWAYS:
+        return getpass(prompt_message)
+    if input_key == PROMPT:
+        if os.getenv(env_fallback):
+            return os.getenv(env_fallback)
+        else:
+            return getpass(prompt_message)
+    return input_key
```

Comment:
Is this needed?

Reply:
Which part?

I don't think the second `getpass` call is need. I reworked the control flow to remove it. `input_key` needs to be returned just in case an actual api key is passed in, Eg 

```python
_get_prompt(os.getenv("MY_CUSTOM_GRETEL_URI"))
```

Reply:
Ah ok. Yea was just asking on the final return but that makes sense